### PR TITLE
Handle Juttle-Viz invalid parameter errors

### PR DESCRIPTION
**Pull request by mnibecker**
Like said above, actually report errors when view is feed improper
variables. The best way to do this was by including the parameter check
in the view run promise. A program shoudn't run if a view has invalid
parameters.

Also part of this pr:
- Jettison redux for the View object. It wasn't doing much for us.
- Instantiate views as soon as we receive the `job_start` event from the
  websocket
- Standardized Errors! Add error classes (akin to what we see in
  juttle/juttle).

@go-oleg (I'm also adding some unit tests)

### CODE REVIEW

File: src/utils/error-messages.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default {
+    'JOB-START-ERROR': 'Event \'job_start\' not received as first point from socket. Problem.',
```
Comment (go-oleg) [inline src/utils/error-messages.js]: `s/first point/first message/`

File: src/utils/errors.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            Error.captureStackTrace(this, this.constructor);
+        }
+
+        // this.code = this.code;
+        this.info = info;
+        this.code = code;
+
+        let template = _.template(messages[this.code], {
+            interpolate: /\{\{([^}]*)\}\}/g,
+            variable: 'info'
```
Comment (go-oleg) [inline src/utils/errors.js]: why use `variable: 'info'`?

File: src/utils/errors.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+}
+
+class JobStartError extends BaseError {
+    constructor(info) {
+        super(info, 'JOB-START-ERROR');
+    }
+}
+
+class ViewParamsError extends BaseError {
```
Comment (go-oleg) [inline src/utils/errors.js]: Maybe `s/View/JuttleView/g` where you are referring to juttle views, because "view" and "viewComponent" are pretty generic.

**Comment by go-oleg** [comment]
lgtm besides the view/juttle-view rename for clarity and other little stuff

File: src/utils/errors.js (hunk 2 above)
Comment (mnibecker) [inline src/utils/errors.js]: it speeds up the time it takes to render the template

**Comment by mnibecker** [comment]
fixes juttle/outrigger#112

**Pull request merged**